### PR TITLE
fix(hatchery): improvements

### PR DESCRIPTION
**Pull request by fsamin**
1. Description
1. Related issues
1. About tests
1. Mentions

@ovh/cds

### CODE REVIEW

File: sdk/hatchery/provisionning.go [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+func provisioning(h Interface, provisionDisabled bool, models []sdk.Model) {
+	if provisionDisabled {
+		log.Debug("provisioning> disabled on this hatchery")
+		return
+	}
+
+	for k := range models {
+		// for a shared.infra hatchery, all models are here (group shared.infra or not)
+		// but, a shared.infra hatchery can provision only a shared.infra model
```
Comment (sguiheux) [inline sdk/hatchery/provisionning.go]: delete this

File: sdk/hatchery/starter.go [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+}
+
+func spawnWorkerForJob(h Interface, j workerStarterRequest) (bool, error) {
+	log.Debug("spawnWorkerForJob> %d", j.id)
+	defer logTime(h, fmt.Sprintf("spawnWorkerForJob> %d elapsed", j.timestamp), time.Now())
+	maxProv := h.Configuration().Provision.MaxConcurrentProvisioning
+	if maxProv < 1 {
+		maxProv = defaultMaxProvisioning
+	}
+	if atomic.LoadInt64(&nbWorkerToStart) > int64(maxProv) {
```
Comment (sguiheux) [inline sdk/hatchery/starter.go]: &gte = instead of > ?

File: sdk/hatchery/provisionning.go (hunk 1 above)
Comment (fsamin) [reply sdk/hatchery/provisionning.go]: no

File: sdk/hatchery/pool.go [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+	}
+
+	// Then: get all worker in the orchestrator queue
+	startedWorkers := h.WorkersStarted()
+	if err != nil {
+		return nil, fmt.Errorf("unable to get started workers: %v", err)
+	}
+
+	// Make the union of the two slices
+	allWorkers := []sdk.Worker{}
```
Comment (bnjjj) [inline sdk/hatchery/pool.go]: it seems you can prealloc a little bit

**Pull request merged**